src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -314,6 +290,110 @@ VectorDims LinearIR::get_master_shape() const {
     return master_shape;
 }
 
+LinearIR::exprIt LinearIR::insert_node(const std::shared_ptr<ov::Node>& new_node, const std::vector<PortConnectorPtr>& new_inputs,
+                                       const std::vector<size_t>& loop_ids, bool update_loop_ports, const constExprIt& place,
+                                       const std::vector<std::set<ExpressionPort>>& consumers) {
+    const auto new_expr = create_expression(new_node, new_inputs);
+    new_expr->set_loop_ids(loop_ids);
+
+    OPENVINO_ASSERT(consumers.empty() || consumers.size() == new_expr->get_output_count(),
+                    "Failed to insert node: count of consumer sets must be sero or equal to output port count");
+    for (size_t i = 0; i < consumers.size(); ++i) {
+        const auto& port_consumers = consumers[i];
+        replace_input_port_connectors(port_consumers, new_expr->get_output_port_connector(i));
+    }
+
+    if (update_loop_ports) {
+        m_loop_manager->update_loop_ports_by_inserted_expr(new_expr);
+    }
+
+    return insert(place, new_expr);
+}
+
+LinearIR::exprIt LinearIR::insert_node(const std::shared_ptr<ov::Node>& new_node, const std::vector<size_t>& loop_ids, bool update_loop_ports,
+                                       const constExprIt& place, const std::vector<std::set<ExpressionPort>>& consumers) {
+    std::vector<PortConnectorPtr> new_inputs(new_node->get_input_size());
+    for (size_t i = 0; i < new_node->get_input_size(); ++i) {
+        const auto& source = new_node->get_input_source_output(i);
+        new_inputs[i] = get_expr_by_node(source.get_node_shared_ptr())->get_output_port_connector(source.get_index());
+    }
+    return insert_node(new_node, new_inputs, loop_ids, update_loop_ports, place, consumers);
+}
+
+LinearIR::exprIt LinearIR::insert_node(const std::shared_ptr<ov::Node>& new_node, const std::vector<size_t>& loop_ids, bool update_loop_ports,
+                                       const constExprIt& place, const std::set<ExpressionPort>& consumers) {
+    const auto consumers_py_port = consumers.empty() ? std::vector<std::set<ExpressionPort>>{} : std::vector<std::set<ExpressionPort>>{ consumers };
+    return insert_node(new_node, loop_ids, update_loop_ports, place, consumers_py_port);
+}
+
+LinearIR::exprIt LinearIR::replace_node(const std::shared_ptr<ov::Node>& new_node, const std::vector<ExpressionPtr>& old_exprs,
+                                        const std::vector<size_t>& loop_ids, const constExprIt& place) {
+    OPENVINO_ASSERT(!old_exprs.empty(), "Failed to replace node: there are no old expressions for replacing");
+     OPENVINO_ASSERT(new_node->get_output_size() == old_exprs.back()->get_output_count(),
```

Comment:
Minor: extra indentation
```suggestion
    OPENVINO_ASSERT(new_node->get_output_size() == old_exprs.back()->get_output_count(),
```

Reply:
Sure, thanks